From ff060cc2092b2ce5cd3253fa5b64aaeeb6464fab Mon Sep 17 00:00:00 2001
From: Tim Yao <tim.yao@amlogic.com>
Date: Wed, 1 Apr 2015 11:56:51 -0700
Subject: [PATCH 5523/5965] PD#105005: amports: improve FF/FB last frame
 keeping.

Wait until RDMA staging settings finish before
saving last video frame.

Change-Id: I26d565cac504a1b89849ae40793178a7cb039b59
---
 drivers/amlogic/amports/video.c | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/drivers/amlogic/amports/video.c b/drivers/amlogic/amports/video.c
index b37611b4aa45..3df5b9fd8057 100755
--- a/drivers/amlogic/amports/video.c
+++ b/drivers/amlogic/amports/video.c
@@ -4048,6 +4048,14 @@ unsigned int vf_keep_current(void)
     }
 #endif
     cur_index = READ_VCBUS_REG(VD1_IF0_CANVAS0 + cur_dev->viu_off);
+
+#ifdef CONFIG_VSYNC_RDMA
+    if (cur_index != disp_canvas[rdma_canvas_id][0]) {
+        msleep(60);
+        cur_index = READ_VCBUS_REG(VD1_IF0_CANVAS0 + cur_dev->viu_off);
+    }
+#endif
+
     y_index = cur_index & 0xff;
     u_index = (cur_index >> 8) & 0xff;
     v_index = (cur_index >> 16) & 0xff;
-- 
2.19.0

